### DISCUSSION TITLE
Create a negative / not in numeric range filter

### DISCUSSION THREAD

**Discussion by anthonyrentalescapes** [algolia/instantsearch#6175] [Q&A]
Hey! I'm wondering how I would go about creating a negative range filter component for example filter:!=[123..1245]

I see there is exclude refinement but I don't see anything with excluding a range

**Comment by Haroenv** [comment]
We don't have that feature, so you'll have to invert the range yourself first, and set a custom numeric refinement with the opposite value:

```js
numericRefinements: {
  price: {
    '<=': 123,
    '>=': 1245;
  }
}
```

**Reply by anthonyrentalescapes** [reply]
thank you for the reply! If I wanted to implement this where would should I start? unfortunately reversing is slower

**Reply by Haroenv** [reply]
If you fork connectRange, and swap the operators in https://github.com/algolia/instantsearch/blob/b3139ed362b92247730fbccd5802a10b6f887214/packages/instantsearch.js/src/connectors/range/connectRange.ts#L236-L250, that should do what you expect I believe

**Reply by anthonyrentalescapes** [reply]
hey yes thank you that does work!
If I want to combine that and do something like filter_has_x:false OR (filter:<=x  and filter:=>=y)
I also want to include results that don't have values in that filter array

**Reply by Haroenv** [reply]
you can't mix disjunctive of filters of a different type, I'm sorry. What you could do however is replace the `filter_has_x` with some value outside of the normal range, like 0, so that it's always included in the other filters already.

In general I think it may be interesting to invert your data so you can use the regular range widget though, but I can see how that's harder for your

**Reply by anthonyrentalescapes** [reply]
Ok I understand. Is there another approach that could allow me to have a custom filter_by set during the request? I appreciate the pointers there's a lot to unpack with how the widgets tie into the api.

**Reply by Haroenv** [reply]
Not fully sure what you mean, but the hook you can use is similar to https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/js/, but you modify the parameters before calling the search function
